src/figuresClasses.ts

```diff
@@ -1,11 +1,85 @@
-export interface Figure {}
+type Color = 'red' | 'green' | 'blue';
+type Shape = 'triangle' | 'circle' | 'rectangle';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  color: Color;
+  shape: Shape;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error(
+        'The side of the figure cannot be less than or equal to zero',
+      );
+    }
+
+    const maxLength: number = Math.max(a, b, c);
+    const totalLength: number = a + b + c;
+
+    if (maxLength >= totalLength - maxLength) {
+      throw new Error("Sides 1, 2, and 3 can't form a triangle");
```

Comment:
you're using this message from instruction but numbers 1, 2, 3 are used because those were the values given as sides. It would be better to substitute the values of sides as a, b, c
```suggestion
      throw new Error(`Sides ${a}, ${b}, and ${c} can't form a triangle`);
```
![image](https://github.com/user-attachments/assets/5d3bf764-37de-4e63-a95c-cc682a741185)